parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -400,17 +400,32 @@ function nodePrep {
             logs_to_events "AKS.CSE.ensureMigPartition" ensureMigPartition
         fi
 
+        # Install GPU device plugin if needed and not already installed
+        if [ "${ENABLE_GPU_DEVICE_PLUGIN_IF_NEEDED}" = "true" ]; then
+            if ! systemctl list-unit-files | grep -q "nvidia-device-plugin.service"; then
+                echo "Installing nvidia-device-plugin package..."
+                logs_to_events "AKS.CSE.installNvidiaDevicePlugin" "installNvidiaDevicePluginPkgFromCache"
+            else
+                echo "nvidia-device-plugin package already installed"
+            fi
+        fi
+
         # Configure GPU device plugin
         if [ "${ENABLE_GPU_DEVICE_PLUGIN_IF_NEEDED}" = "true" ]; then
```

Comment:
why two different if clause with the same comparaison ? line 414 and 404 should be combiened no ?